src/test/components/ServiceWorkerManager.test.js

```diff
@@ -91,69 +106,178 @@ describe('app/ServiceWorkerManager', () => {
+
+      installOptions.onUpdateReady();
+      expect(serviceWorkerRuntime.applyUpdate).not.toHaveBeenCalled();
+      expect(container.firstChild).toBe(null);
+
+      await dispatch(viewProfile(_getSimpleProfile()));
+      expect(container.firstChild).not.toBe(null);
+    });
+
+    it(`doesn't show a notice until we're done symbolicating`, async () => {
+      process.env.NODE_ENV = 'production';
+
+      const { navigateToStoreLoadingPage, dispatch, container } = setup();
+
+      navigateToStoreLoadingPage();
+      await dispatch(viewProfile(_getSimpleProfile()));
+      dispatch(startSymbolicating());
+
+      const installOptions = serviceWorkerRuntime.install.mock.calls[0][0];
+
+      installOptions.onUpdateReady();
+      expect(serviceWorkerRuntime.applyUpdate).not.toHaveBeenCalled();
+      expect(container.firstChild).toBe(null);
+
+      dispatch(doneSymbolicating());
+      expect(container.firstChild).not.toBe(null);
+    });
+
+    it('shows a warning notice if the SW was updated before we were ready', async () => {
+      process.env.NODE_ENV = 'production';
 
-    const { navigateToStoreLoadingPage, container } = setup();
-    navigateToStoreLoadingPage();
+      const { navigateToStoreLoadingPage, dispatch, container } = setup();
 
-    const installOptions = serviceWorkerRuntime.install.mock.calls[0][0];
-    installOptions.onUpdated();
-    expect(container.firstChild).not.toBe(null);
+      navigateToStoreLoadingPage();
+      await dispatch(viewProfile(_getSimpleProfile()));
+      dispatch(startSymbolicating());
+
+      const installOptions = serviceWorkerRuntime.install.mock.calls[0][0];
+
+      installOptions.onUpdateReady();
+      expect(serviceWorkerRuntime.applyUpdate).not.toHaveBeenCalled();
+      expect(container.firstChild).toBe(null);
+
+      // Some other tabs updated the SW.
+      installOptions.onUpdated();
+      expect(
+        ensureExists(container.querySelector('.photon-message-bar')).className
+      ).toMatch(/\bphoton-message-bar-warning\b/);
+      expect(container.firstChild).toMatchSnapshot();
+    });
   });
 
-  it('does not show a notice with the `from-addon` datasource', () => {
-    process.env.NODE_ENV = 'production';
+  describe('with the `from-addon` datasource', () => {
+    it(`doesn't show a notice if updated after we were fully loaded`, async () => {
+      process.env.NODE_ENV = 'production';
+
+      const { navigateToAddonLoadingPage, container, dispatch } = setup();
+      navigateToAddonLoadingPage();
+      await dispatch(viewProfile(_getSimpleProfile()));
+
+      const installOptions = serviceWorkerRuntime.install.mock.calls[0][0];
+
+      // We don't display anything if there's an update ready.
+      installOptions.onUpdateReady();
+      expect(serviceWorkerRuntime.applyUpdate).not.toHaveBeenCalled();
+      expect(container.firstChild).toBe(null);
+
+      // And we still don't if it was updated elsewhere.
```

Comment:
This is the case where I'm not sure about the ideal behavior. What do you think?

It should be quite straightforward to change later if we feel it's better.